### ISSUE TITLE
[Refactoring proposal] Maybe `IntlMessageFormat.__addLocaleData(` could be omitted

### ISSUE THREAD

**Issue by catamphetamine** [formatjs/intl-messageformat#130]
While browsing
https://github.com/yahoo/intl-messageformat/tree/master/dist/locale-data
I noticed `IntlMessageFormat.__addLocaleData(` everywhere.
Maybe it could be refactored out of these files?

**Comment by ericf** [comment]
Yes, ideally the locale data modules should just _provide_ the locale data. After refactoring React Intl, that's what I switched to. Now the locale data files look like this: https://npmcdn.com/react-intl@2.0.1/locale-data/af.js

It's minified, but the locale data has a UMD wrapper and will only expose itself on the global `ReactIntlLocaleData` object if the runtime it's being loaded in doesn't have a module system.

**Comment by halt-hammerzeit** [comment]
Ok, that would do the trick.
The reason I was asking is because of this use case:
https://github.com/halt-hammerzeit/javascript-time-ago#usage

Not an issue really, just a couple of lines of `global` tricks.

```
global.IntlMessageFormat = require('javascript-time-ago/node_modules/intl-messageformat')
require('javascript-time-ago/node_modules/intl-messageformat/dist/locale-data/en')
require('javascript-time-ago/node_modules/intl-messageformat/dist/locale-data/ru')
delete global.IntlMessageFormat
```

This issue can be closed since it's in your todo list.

**Comment by brettz9** [comment]
In issue #174, I am requesting the locale files be converted into ES6 modules so that the benefits mentioned by @ericf could be available to those using Rollup or what not to convert to UMD but it'd also work for those wishing to convert any combination of locale files to browser-usable ES modules or whatever they wish (while source could rely on fully standard ES).

**Comment by longlho** [comment]
fixed in #188
